src/usr/local/www/services_dhcpv6.php

```diff
@@ -313,8 +313,11 @@ function dhcpv6_apply_changes($dhcpdv6_enable_changed) {
 	if ($_POST['ddnsupdate'] && !is_domain($_POST['ddnsdomain'])) {
 		$input_errors[] = gettext("A valid domain name must be specified for the dynamic DNS registration.");
 	}
-	if ($_POST['ddnsupdate'] && !is_ipaddrv4($_POST['ddnsdomainprimary'])) {
-		$input_errors[] = gettext("A valid primary domain name server IPv4 address must be specified for the dynamic domain name.");
+	if ($_POST['ddnsupdate'] && !is_ipaddrv6($_POST['ddnsdomainprimary'])) {
```

Comment:
Does this have to be IPv6? It's a DNS server address, it shouldn't have to match the type of address the daemon is handling. It should probably accept either type in either page (IPv4 or IPv6 on DHCPv4 and DHCPv6). May need testing to confirm that it works in both of those cases.

Reply:
True,
it uses primary6 and secondary6 for IPv6 addresses,
see dhcpd.conf(5):
```
       The primary statement specifies the IP address of the name server whose
       zone  information  is to	be updated.  In	addition to the	primary	state-
       ment there are also the primary6	, secondary and	secondary6 statements.
       The  primary6  statement	specifies an IPv6 address for the name server.
```